engine/include/Engine/Core/TimeSystem/TimeSystem.hpp

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2021 Amara Sami, Dallard Thomas, Nardone William, Six Jonathan
+ * This file is subject to the LGNU license terms in the LICENSE file
+ *	found in the top-level directory of this distribution.
+ */
+
+#pragma once
+
+#include "Engine/Core/Debug/Log.hpp"
+#include <chrono>     // std::chrono::seconds, std::chrono::system_clock::now()
+#include <functional> //std::function
+#include <queue>
+
+namespace Engine::Core
+{
+struct TimerTask
+{
+    double localTimer = 0.; // if current time egal 1s and local timer egal 0.5 global time egal 1.5
+    double globalTimer = 0.;
+    std::function<void()> task = nullptr;
+    bool isLooping = false;
+
+    bool operator>(const TimerTask& other) const noexcept
+    {
+        return globalTimer > other.globalTimer;
+    }
+};
+
+class TimeSystem
+{
+protected:
+    std::chrono::system_clock::time_point m_time = std::chrono::system_clock::now();
+    std::chrono::system_clock::time_point m_tempTime = m_time;
+    double m_fixedTimeAcc = 0.;
+
+    unsigned int m_frameAcc = 0;
```

Comment:
`m_frameAcc` is not used anywhere.
If it is meant to be used, move it after the `double`s to align it properly